src/Symfony/Component/HttpKernel/DependencyInjection/RegisterControllerArgumentLocatorsPass.php

```diff
@@ -120,6 +121,21 @@ public function process(ContainerBuilder $container)
                 $args = [];
                 foreach ($parameters as $p) {
                     /** @var \ReflectionParameter $p */
+                    if ($attr = $p->getAttributes(Autowire::class)[0] ?? null) {
+                        $value = $attr->newInstance()->value;
```

Comment:
class_exists was needed because we create an instance here ;)

Reply:
Doesn't the above if statement evaluate to false if `Autowire` doesn't exist?

Reply:
Nope, attributes being declarative, their implantation is not required until one calls newInstance

Reply:
TIL :)